Cargo.toml

```diff
@@ -21,14 +21,30 @@ use-serde = ["serde", "bitcoin_hashes/serde", "secp256k1/serde"]
 secp-lowmemory = ["secp256k1/lowmemory"]
 secp-recovery = ["secp256k1/recovery"]
 
+# either std or no-std must be selected
+std = ["secp256k1/std", "bitcoin_hashes/std", "bech32/std"]
+no-std = ["hashbrown/nightly", "core2/alloc", "bitcoin_hashes/alloc"]
+
+compact-block-filters = ["hashbrown"]
```

Comment:
I'm not really a big fan of using `hashbrown` for all our hashmaps on std just because we use it on no_std. Can we just always compile compact-block-filters if `hashbrown || std`? That would avoid polluting the feature space as well.

Reply:
hmm... not sure what I was thinking there.  I removed the need for hashbrown if std is on.

Also, I made compact-block-filters and merkleblocks compile unconditionally, and grab the hashmap from either std or hashbrown.

Does this sound good?